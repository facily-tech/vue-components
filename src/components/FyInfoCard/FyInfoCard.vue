<template>
  <div class="fy-info-card d-flex align-center justify-center info px-5 py-5">
    <img :src="img" alt="John" height="50" />

    <div class="flex-column justify-space-between ml-5 text-right">
      <div class="text-h5 font-weight-medium white--text mb-5">{{ title }}</div>

      <fy-button color="info">{{ caption }}</fy-button>
    </div>
  </div>
</template>

<script lang="ts">
import Vue from 'vue';

import { FyButton } from '@/index';

export default Vue.extend({
  name: 'fy-info-card',

  components: {
    FyButton,
  },

  props: {
    img: { type: String, default: '' },
    title: { type: String, default: '' },
    caption: { type: String, default: '' },
  },
});
</script>

<style lang="scss" scoped>
.fy-info-card {
  width: 420px;
  height: 240px;
  border-radius: 4px;
}
</style>
