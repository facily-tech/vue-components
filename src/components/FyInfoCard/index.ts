export { default as FyInfoCard } from './FyInfoCard.vue';
