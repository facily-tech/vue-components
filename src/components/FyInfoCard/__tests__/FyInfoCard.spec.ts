import Vue from 'vue';
import Vuetify from 'vuetify';
import FyInfoCard from '../FyInfoCard.vue';
import { mount, MountOptions, Wrapper } from '@vue/test-utils';

Vue.use(Vuetify);

describe('FyInfoCard', () => {
  // eslint-disable-line max-statements
  type Instance = InstanceType<typeof FyInfoCard>;
  let mountFunction: (options?: MountOptions<Instance>) => Wrapper<Instance>;
  beforeEach(() => {
    mountFunction = (options?: MountOptions<Instance>) => {
      return mount(FyInfoCard, {
        vuetify: new Vuetify(),
        ...options,
      });
    };
  });

  it('should render component and match snapshot', () => {
    const wrapper = mountFunction();

    expect(wrapper.html()).toMatchSnapshot();
  });
});
