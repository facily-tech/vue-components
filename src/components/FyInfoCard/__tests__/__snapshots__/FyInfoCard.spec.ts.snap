// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FyInfoCard should render component and match snapshot 1`] = `
<div class="fy-info-card d-flex align-center justify-center info px-5 py-5"><img src="" alt="John" height="50">
  <div class="flex-column justify-space-between ml-5 text-right">
    <div class="text-h5 font-weight-medium white--text mb-5"></div> <button type="button" class="v-btn v-btn--has-bg theme--light elevation-0 v-size--default info" color="info"><span class="v-btn__content"></span></button>
  </div>
</div>
`;
